rfcs/tag-component.md

```diff
@@ -49,5 +49,5 @@ export default () => {
 * `overrides: {Root: (props: {[string]: any}) => React$Node, Action: (props: {[string]: any}) => React$Node}`
   * Custom `Action` button.
   * `Root` wrapper element for the whole tag control to apply styles
-* `onActionClick: func(evt => void)`:
-  handler for events on Action button element (default is Cross).
+* `onActionClick: func((evt, children) => void)`:
+  handler for events on Action button element (default is Cross). `children` provides which actual Tag was clicked
```

Comment:
I'm trying to figure out how much value `children` has here. It seems like if it's an actual react element it has zero value. If its a string it has a little bit more value, though usually the tag has the human-readable label like "San Francisco" instead of the key like "san_francisco".

Maybe we should instead encourage users to do something like

```js
tags.map(() => (
  <Tag onActionClick={this.onClose.bind(this, tag.key)}>{tag.label}</Tag>
))
```

Reply:
yes, React element has no sense. String label has sense and was added only to avoid extra efforts by user to find himself, which tag has been clicked actually. I mean, Idk, what would be the most common case for example. Last time (in Select I believe) in similar case I was asked to add extra value in handler to help user not spend time to find Option clicked.